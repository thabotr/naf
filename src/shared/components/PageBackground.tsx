import React, {ReactNode} from 'react';
import {StyleSheet} from 'react-native';
import {Surface} from 'react-native-paper';
import {useThemedStyles} from '../providers/theme';

type Props = {
  children?: ReactNode;
  pageLabel?: string;
  style?: any;
};

export default function PageBackground({children, pageLabel, style}: Props) {
  const styles = useThemedStyles(styleSheet);
  return (
    <Surface accessibilityLabel={pageLabel} style={[styles.background, style]}>
      {children}
    </Surface>
  );
}

const styleSheet = (theme: any) =>
  StyleSheet.create({
    background: {
      width: '100%',
      height: '100%',
      backgroundColor: theme.color.secondary,
    },
  });
