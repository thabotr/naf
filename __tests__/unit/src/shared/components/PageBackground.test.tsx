import React from 'react';
import renderer from 'react-test-renderer';
import PageBackground from '../../../../../src/shared/components/PageBackground';
import themed from '../../../utils/themed';

describe('PageBackground', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(themed(<PageBackground pageLabel="test page" />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
