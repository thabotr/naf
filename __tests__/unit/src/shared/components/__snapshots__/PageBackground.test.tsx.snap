// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageBackground renders correctly 1`] = `
<View
  accessibilityLabel="test page"
  collapsable={false}
  style={
    {
      "backgroundColor": "#dbdbdb",
      "height": "100%",
      "shadowColor": "#000000",
      "shadowOffset": {
        "height": 3,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 4,
      "width": "100%",
    }
  }
/>
`;
